### PR TITLE
Fix #18467: Update FilterSelected logic when the active screen is the Track Manager

### LINKED ISSUE #18467: "Selected only" object Selection filters is active in track design manager, and cannot be toggled
### Operating System

Windows 10 x64

### OpenRCT2 build

OpenRCT2 v0.4.2-109-g2aa46a7 (2aa46a7 on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

No rides are listed when this filter is enabled. This filter cannot be changed, and it makes no sense for the track design manager, so it should be ignored.

### Steps to reproduce

1. Open any park or the scenario editor
2. Tick the filter "Selected only"
3. Open the track design manager
4. Notice how no rides are listed

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by mdbckrtn**
The Track Manager has no concept of selectable items. Therefore we shouldn't be concerned if the SELECTED or NONSELECTED filters are currently active in the global filters list when on this screen and, as such, those filters should be ignored when determining which items to display in the item list.

This PR resolves issue #18467

### CODE REVIEW

**Comment by kurbus** [comment]
dankeshon

File: src/openrct2-ui/windows/EditorObjectSelection.cpp [hunk 1]
```diff
@@ -1307,15 +1307,20 @@ class EditorObjectSelectionWindow final : public Window
...
+        if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
+        {
+            return true;
+        }
+        else if (_FILTER_SELECTED == _FILTER_NONSELECTED)
         {
             return true;
         }
-        if (_FILTER_SELECTED && objectFlag & ObjectSelectionFlags::Selected)
+        else if (_FILTER_SELECTED && objectFlag & ObjectSelectionFlags::Selected)
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/EditorObjectSelection.cpp:1319]: There is no point using else if we’re returning anyway, so please drop it from lines 1315, 1319 and 1323.

Comment (mdbckrtn) [reply src/openrct2-ui/windows/EditorObjectSelection.cpp]: Removed

**Review by Gymnasiast — APPROVED** [review]

**Comment by Gymnasiast** [comment]
Thanks!

Since it was your first contribution, I amended the changelog and reworded the commits for you. Make sure you always mentioned the _bug_ you fixed in your commit message, rather than the solution.

**Pull request merged**